client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts

```diff
@@ -208,92 +169,109 @@ const siteMigration: FlowV2 = {
 							return navigate(
 								paths.sitePickerPath( {
 									from: fromQueryParam,
+									platform: platformQueryParam || 'unknown',
 									...queryParams,
 								} )
 							);
 						}
 						case 'select-site': {
-							const { ID: newSiteId, slug: newSiteSlug } =
-								providedDependencies.site as SiteExcerptData;
+							const { ID: siteId, slug: siteSlug } = providedDependencies.site as SiteExcerptData;
 
 							if ( 'migrate' === actionQueryParam ) {
 								return navigate(
 									paths.howToMigratePath( {
-										siteSlug: newSiteSlug,
-										siteId: newSiteId,
+										siteSlug,
+										siteId,
 										from: fromQueryParam,
 									} )
 								);
 							}
-							return navigate(
-								paths.importOrMigratePath( { siteSlug: newSiteSlug, siteId: newSiteId } )
-							);
+
+							if ( platformQueryParam !== 'wordpress' ) {
+								//TODO: Improve it to redirect the user to the correct import step if the platform is not wordpress
+								return exitFlow(
+									paths.siteSetupImportListPath( {
+										from: fromQueryParam,
+										siteSlug,
+										siteId,
+										backToFlow: `/${ flowPath }/${ STEPS.PICK_SITE.slug }`,
+										origin: '',
+									} )
+								);
+							}
+
+							return navigate( paths.importOrMigratePath( { siteSlug, siteId } ) );
 						}
 						case 'create-site':
-							return navigate( paths.siteCreationPath( { from: fromQueryParam } ) );
+							return navigate(
+								paths.siteCreationPath( { from: fromQueryParam, platform: platformQueryParam } )
+							);
 					}
 				}
 
 				case STEPS.SITE_CREATION_STEP.slug: {
-					const queryArgs = {
-						from: fromQueryParam,
-						skipMigration: 'import' === actionQueryParam ? true : undefined,
```

Comment:
I removed the skipMigration flag because it is possible to infer the same state via the platform= query param.